framework/src/main/java/org/checkerframework/framework/util/SimpleQualifierHierarchy.java

```diff
@@ -0,0 +1,304 @@
+package org.checkerframework.framework.util;
+
+import java.lang.annotation.Annotation;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.util.Elements;
+import org.checkerframework.framework.type.QualifierHierarchy;
+import org.checkerframework.framework.util.QualifierKindHierarchy.QualifierKind;
+import org.checkerframework.javacutil.AnnotationBuilder;
+import org.checkerframework.javacutil.AnnotationUtils;
+import org.checkerframework.javacutil.BugInCF;
+import org.checkerframework.javacutil.TypeSystemError;
+
+/**
+ * A qualifier hierarchy where no qualifiers have arguments; that is, no qualifier is represented by
+ * an annotation with elements. The meta-annotation {@link
+ * org.checkerframework.framework.qual.SubtypeOf} specifies the hierarchy.
+ *
+ * <p>It uses a {@link QualifierKindHierarchy} to model the relationships between qualifiers.
+ * Subclasses can override {@link #createQualifierKindHierarchy(Collection)} to return a subclass of
+ * QualifierKindHierarchy.
+ */
+public class SimpleQualifierHierarchy extends QualifierHierarchy {
```

Comment:
I'm still thinking on this.  
SimpleQualifierHierarchy:
* NoElementsQualifierHierarchy
* QualifierHierarchyWithElements
* DefaultQualifierHierarchy (Because it's the QualifierHierarchy used by default.)

ComplexQualifierHierarchy:
* ElementsQualifierHierarchy
* QualifierHierarchyWithElements
* QualifierWithValuesHierarchy

Reply:
The suffixed versions QualifierHierarchyWithoutElements and QualifierHierarchyWithElements read best to me, among the options you list.